src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -1839,24 +1839,47 @@ trait Types
   class PackageClassInfoType(decls: Scope, clazz: Symbol)
   extends ClassInfoType(List(), decls, clazz)
 
-  /** A class representing a constant type.
+  /** A class representing a constant type. Constant types are constant-folded during type checking.
+   *  To avoid constant folding, use the type returned by `deconst` instead.
```

Comment:
Could you expand the documentation here and explain why we need the distinciton between `FoldableConstantType` and `LiteralType`? Which of the two is used when / where?

Reply:
Will do.